### ISSUE TITLE
Add checkbox to toggle word wrapping

### ISSUE THREAD

**Issue by waldyrious** [browserpad/browserpad#50]
Similar to the spell-checking one.

**Comment by SenadAlagic** [comment]
I'd like to help here but I am confused. What would it mean if the function were on and what would it mean if the function is off?

**Comment by waldyrious** [comment]
Good question — this should definitely be documented. :sweat_smile:

Currently the main textarea has a width of 100% of available horizontal space, and wraps text by adding a bunch of padding to the right:

https://github.com/browserpad/browserpad/blob/18894b472f47e2be51a801118b3f9f9f818cedce/index.xhtml#L22

...so that the space left out for text equals the current hard-coded line-wrapping limit of 80 characters: 

https://github.com/browserpad/browserpad/blob/18894b472f47e2be51a801118b3f9f9f818cedce/index.xhtml#L27

I believe simply removing the extra padding and adding `white-space: pre;` to the main textarea (as well as hiding the `linewidth-guide` auxiliary textarea) might be sufficient to implement this.

**Comment by waldyrious** [comment]
To be honest, I don't remember why I implemented the line-wrapping (back in ecbb3f0f04e6b0b558ba72257e30d5e5bb734ac7) using a padding rather than by changing the width of the textarea itself :thinking:

For context, there is some information in #17 and in the comments of 62264d8354bce3cdc0ff9d46d8c1fbf6de54ca0f that perhaps could be useful, but if not, I'd be open to experimenting with implementing wrapping by editing the width of the textarea, rather than its padding.

**Comment by waldyrious** [comment]
Another thing related to this functionality that might be improved is using a CSS background to display the line-wrapping guide, rather than the right border of the extra textarea introduced back in 25d8c64cc0801802a8bb24a465bc836a8acb59b5. Perhaps something similar to what was suggested [here](https://github.com/refined-github/refined-github/issues/1484#issuecomment-562652779) might work.

**Comment by SenadAlagic** [comment]
Oh I understand what you wanted to do! I'll do it right now, no problems.
This I don't quite understand.
__Another thing related to this functionality that might be improved is using a CSS background to display the line-wrapping guide, rather than the right border of the extra textarea introduced back in https://github.com/browserpad/browserpad/commit/25d8c64cc0801802a8bb24a465bc836a8acb59b5__

**Comment by waldyrious** [comment]
Don't bother about that bit. I should have opened a separate issue, and will do so now. :)

**Comment by SenadAlagic** [comment]
Oh okay. I'm having some trouble with manipulating CSS with JS (CSS in general gives me troubles) but this should be fixed soon!

**Comment by SenadAlagic** [comment]
Hello! I completed the issue but I can't seem to create a pull request while another one is active? I'm sorry for your troubles.

**Comment by waldyrious** [comment]
> Hello! I completed the issue

That was fast! :grinning:

> I can't seem to create a pull request while another one is active

It's probably better to start [using separate branches](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches) for each feature; that way you will be able to submit multiple PRs simultaneously. Otherwise, all changes you make to the `gh-branch` will apply to the PR you already have open.

> I'm sorry for your troubles.

No worries at all! I'm happy to have someone else contributing to the project :)

**Comment by SenadAlagic** [comment]
> It's probably better to start [using separate branches](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches) for each feature; that way you will be able to submit multiple PRs simultaneously. Otherwise, all changes you make to the `gh-branch` will apply to the PR you already have open.
> 
Yes I sort of figured it out later but it's too late now? I guess two issues are fixed in the same PR now?

**Comment by waldyrious** [comment]
Yeah, no problem — you can do that for the next PR. Once #58 is merged, I recommend you reset your gh-pages branch as I described [here](https://github.com/browserpad/browserpad/issues/42#issuecomment-1126895215), and then start using separate branches for future PRs.

**Comment by SenadAlagic** [comment]
I'm sorry as this too will be off topic but I have questions. Say I checkout a new branch and write some code (I fix an issue for ex) and I commit and push. When you merge do I need to checkout the gh-pages and "git pull" or? I assume that's what it would be for. And then I assume I delete the branch I worked on? It's kind of confusing 😄

**Comment by waldyrious** [comment]
Yeah, it is confusing indeed. But you're in the right track. I will quote what you wrote and add additional text (in bold) to make it clear what I understood from it:

> Say I checkout a new branch and write some code (I fix an issue for ex) and I commit and push **and create a pull request from it**. When you merge **the pull request,** do I need to checkout the gh-pages and "git pull" or? I assume that's what it would be for. And then I assume I delete the branch I worked on?

If this is what you meant, then yes, that's exactly right!

By the way, you may want to check out these two resources that are excellent at providing intuition for how git works :)

- https://learngitbranching.js.org/
- https://www.youtube.com/watch?v=1ffBJ4sVUb4

**Comment by RageGamerBoi** [comment]
> To be honest, I don't remember why I implemented the line-wrapping (back in [ecbb3f0](https://github.com/browserpad/browserpad/commit/ecbb3f0f04e6b0b558ba72257e30d5e5bb734ac7)) using a padding rather than by changing the width of the textarea itself 🤔

Probably because the scrollbar in the textbox messed things up. It took a fair bit of math but I’ve dealt with that in #75
